src/cargo/ops/cargo_rustc/custom_build.rs

```diff
@@ -398,7 +399,8 @@ pub fn build_map<'b, 'cfg>(cx: &mut Context<'b, 'cfg>,
         // linkable to us (e.g. not a binary) and it's for the same original
         // `kind`.
         let mut ret = Vec::new();
-        for &(pkg, target, p) in cx.dep_targets(pkg, target, profile).iter() {
+        for &(pkg, target, p) in cx.dep_targets(pkg, target, Kind::Host, profile)
```

Comment:
I also think that this needs to be `kind` instead of `Kind::Host`. It'd be great to make sure there's a test requiring this, and I *think* it'll be triggered with a build script on a package that has a target-specific dependency. Currently that target-specific dependency's variables won't show up for the build script, but if this is `kind` I think they will (although I could be wrong here)